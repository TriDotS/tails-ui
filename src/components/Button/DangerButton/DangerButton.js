import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

const Button = styled.button`
  &:focus {
    outline: none;
  }
`

const DangerButton = ({ children, outline, fullWidth }) => (
  <Button
    className={
      outline
        ? `${fullWidth && 'w-full'}bg-transparent hover:bg-red text-red-dark font-semibold hover:text-white py-2 px-4 border-2 border-red hover:border-transparent rounded`
        : `${fullWidth && 'w-full'}bg-red hover:bg-red-dark text-white font-sans font-bold py-2 px-4 shadow rounded`
    }
  >
    {children}
  </Button>
)

DangerButton.propTypes = {
  children: PropTypes.node,
  outline: PropTypes.bool,
  fullWidth: PropTypes.bool
}

export default DangerButton
