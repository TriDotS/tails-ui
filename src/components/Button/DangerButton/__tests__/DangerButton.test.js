import DangerButton from '../DangerButton'
import React from 'react'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(<DangerButton />, div)
})

test('danger button snapshot', () => {
  const tree = renderer.create(<DangerButton />).toJSON()
  expect(tree).toMatchSnapshot()
})
