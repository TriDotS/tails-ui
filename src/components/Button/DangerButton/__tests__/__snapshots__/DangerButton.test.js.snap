// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`danger button snapshot 1`] = `
<button
  className="undefinedbg-red hover:bg-red-dark text-white font-sans font-bold py-2 px-4 shadow rounded sc-bdVaJa igkfrU"
/>
`;
